code/modules/projectiles/guns/projectile/constructable/siegecannon.dm

```diff
@@ -0,0 +1,221 @@
+/obj/structure/siege_cannon
+	name = "Siege Cannon"
+	desc = "A heavy-duty cannon. Capable of knocking down walls and fortifications when loaded with the right munitions."
+	icon = 'icons/obj/siege_cannon.dmi'
+	icon_state = "siege_cannon"
+	density = TRUE
+	var/obj/item/loadedItem = null
+	var/mob/living/loadedMob = null
+	var/wFuel = 0
+	var/maxFuel = 20
+	var/maxSize = W_CLASS_LARGE	//Anything bigger won't fit.
+
+/obj/structure/siege_cannon/Destroy()
+	unloadCannon()
+	..()
+
+/obj/structure/siege_cannon/proc/unloadCannon()
+	if(loadedItem)
+		loadedItem.forceMove(loc)
+		loadedItem = null
+	if(loadedMob)
+		loadedMob.forceMove(loc)
+		loadedMob = null
+
+/obj/structure/siege_cannon/AltClick(var/mob/user)
+	if(user.incapacitated() || !in_range(user, src) || user.loc == src)
+		return
+	to_chat(user,"<span class='notice'>You unload the [src].</span>" )
+	unloadCannon()
+
+/obj/structure/siege_cannon/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/reagent_containers))
+		fillCannon(W, user)
+		return 1
+	if(W.is_wrench())
+		wrenchAnchor(user, W, 5)	//Half a second to wrench. Being able to turn it via verb while anchored is intentional.
+	else
+		loadCannon(W, user)
+
+/obj/structure/siege_cannon/attack_hand(mob/user)
```

Comment:
`attack_hand` is called for every kind of mob, including ghosts and dead people. While there is sanity for a human missing his hands or being restrained, there is *no* sanity for a dead human or a ghost. Add a stat check.